### PR TITLE
feat: Add specific fuzz tests for cast and try_cast and fix NPE found during fuzz testing

### PR DESCRIPTION
**Pull request by andygrove**
## Which issue does this PR close?

N/A

## Rationale for this change

Ongoing improvements to fuzz testing.

## What changes are included in this PR?

- Add specific tests for `cast` and `try_cast`
- Improve error handing in fuzz testing tool
- Fix an NPE found during fuzz testing

## How are these changes tested?

Manual testing

### CODE REVIEW

File: spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala [hunk 1]
```diff
@@ -2169,10 +2169,10 @@ object QueryPlanSerde extends Logging with ShimQueryPlanSerde with CometExprShim
         val trimCast = Cast(trimStr.get, StringType)
         val trimExpr = exprToProtoInternal(trimCast, inputs)
         val optExpr = scalarExprToProto(trimType, srcExpr, trimExpr)
-        optExprWithInfo(optExpr, expr, null, srcCast, trimCast)
+        optExprWithInfo(optExpr, expr, srcCast, trimCast)
       } else {
         val optExpr = scalarExprToProto(trimType, srcExpr)
-        optExprWithInfo(optExpr, expr, null, srcCast)
```
Comment (andygrove) [inline spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala:2175]: This fixes an NPE that was discovered during fuzz testing

File: spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala [hunk 2]
```diff
@@ -2169,10 +2169,10 @@ object QueryPlanSerde extends Logging with ShimQueryPlanSerde with CometExprShim
...
+        optExprWithInfo(optExpr, expr, srcCast)
```
Comment (parthchandra) [inline spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala:2175]: Thanks for finding and fixing this Andy!

**Review by parthchandra — APPROVED** [review]
LGTM

**Comment by andygrove** [comment]
@kazuyukitanimura @huaxingao Could I get a committer review please

**Review by huaxingao — APPROVED** [review]
LGTM. Thanks for the PR!

**Review by viirya — COMMENTED** [review]
Looks good.

**Pull request merged**